### PR TITLE
refactor(banner): convert to mixins

### LINKED ISSUE #1081: (Opt-in) Convert the Banner
https://cashmere.healthcatalyst.net/guides/using-customizing-components

### PR DESCRIPTION
**Pull request by isaaclyman**
Fixes #1081.

### CODE REVIEW

File: projects/cashmere/src/lib/banner/hc-banner.component.ts [hunk 1]
```diff
@@ -11,7 +11,6 @@ import {parseBooleanAttribute} from '../util';
 })
 export class HcBannerComponent {
     private _type: string = 'info';
-    private _fixedTop: boolean = false;
```
Comment (isaaclyman) [inline projects/cashmere/src/lib/banner/hc-banner.component.ts]: VS Code indicated that this property wasn't being used anywhere, so I deleted it. See also the `fixed-top` class below.

File: projects/cashmere/src/lib/banner/hc-banner.scss [hunk 2]
```diff
@@ -1,110 +1,57 @@
...
-    display: flex;
-    align-items: center;
-    justify-content: flex-start;
-    width: 100%;
-    padding: 10px;
-    z-index: $zindex-subnav;
+    @include hc-banner();
+}
 
-    &.fixed-top {
```
Comment (isaaclyman) [inline projects/cashmere/src/lib/banner/hc-banner.scss]: This `fixed-top` class isn't referenced anywhere in Cashmere (documentation, examples, etc.). It appears to be orphaned. I deleted it, but I can add it back if we believe it's in use.

File: projects/cashmere/src/lib/banner/hc-banner.component.ts (hunk 1 above)
Comment (andrew-frueh) [reply projects/cashmere/src/lib/banner/hc-banner.component.ts]: hey @isaaclyman - you're right about this one.  I believe it was added but then we realized it didn't play nice with subnav.  There's an issue logged to make `fixedTop` a directive that can be used across components.  I'll bump that one up on the priority list.  Bump for this component, we're good to remove that orphaned variable

Comment (andrew-frueh) [reply projects/cashmere/src/lib/banner/hc-banner.component.ts]: This is the issue #654

**Pull request merged**